File: backend/src/main/java/info/gamewise/lor/videos/deckcodeextractor/AbstractDeckCodeExtractor.java
package info.gamewise.lor.videos.deckcodeextractor;

import java.util.Collection;
import java.util.List;
import java.util.stream.Stream;

import static java.util.stream.Collectors.toUnmodifiableList;

abstract class AbstractDeckCodeExtractor {

    protected Stream<String> descriptionAsStream(String description) {
        List<String> splitBySpace = List.of(description.split(" "));
        return splitBySpace.stream()
                .map(this::splitByEnter)
                .collect(toUnmodifiableList())
                .stream()
                .flatMap(Collection::stream);
    }

    private List<String> splitByEnter(String text) {
        return List.of(text.split("\n"));
    }
}

File: backend/src/main/kotlin/info/gamewise/lor/videos/deckcodeextractor/AbstractDeckCodeExtractor.kt
package info.gamewise.lor.videos.deckcodeextractor

abstract class AbstractDeckCodeExtractor {

    protected fun descriptionAsStream(description: String): List<String> {
        val splitBySpace = listOf(*description.split(" ".toRegex()).toTypedArray())
        return splitBySpace
            .map { splitByEnter(it) }
            .flatten()
    }

    private fun splitByEnter(text: String): List<String> {
        return listOf(*text.split("\n".toRegex()).toTypedArray())
    }
}